streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -196,15 +176,14 @@ public void shouldCleanUpTaskStateDirectoriesThatAreNotCurrentlyLocked() throws
             directory.directoryForTask(new TaskId(2, 0));
 
             List<File> files = Arrays.asList(appDir.listFiles());
-            assertEquals(3, files.size());
+            assertEquals(1, files.size());
```

Comment:
Why this? Shouldn't `directory.lock(task0);` and `directory.lock(task1);` have create a lock file each?

Reply:
Will dig some more on these tests once Guozhang confirms the plan.

Reply:
This was due to the specification of StandardOpenOption.DELETE_ON_CLOSE to FileChannel.open call.

Reply:
@mjsax 
I want to get your opinion on whether StandardOpenOption.DELETE_ON_CLOSE should be kept in the PR.
This would affect how test is modified.

Reply:
Sorry for late reply. It was a little crazy the last weeks and I did not find time earlier.

I cannot remember why we want to add the DELETE_ON_CLOSE option? Can you refresh my mind?

Also, I am not sure why this option reduced the file count? I understand that the task directories are actually not created any longer, however, we moved both lock files up the hierarchy and thus the count should not change?

Also, did you see this older comment: https://github.com/apache/kafka/pull/5650#issuecomment-426840918 For a clean upgrade path, addressing this issue is important.